arbiter.c

```diff
@@ -667,33 +535,42 @@ void ARB_Clock(arbiter_t *chip)
 
-    chip->w91 = (chip->input.ext_vaddress_in & 0x180) == 0;
-    chip->w99 = (chip->input.ext_vaddress_in & 0x180) == 0x80;
-    chip->w90 = (chip->input.ext_vaddress_in & 0x180) == 0x100;
-    chip->w92 = (chip->input.ext_vaddress_in & 0x180) == 0x180;
+    chip->w91 = 0;
+    chip->w99 = 0;
+    chip->w90 = 0;
+    chip->w92 = 0;
+
+    switch (chip->input.ext_vaddress_in & 0x180) {
+        case 0:
+            chip->w91 = 1;
+            break;
+        case 0x80:
+            chip->w99 = 1;
+            break;
+        case 0x100:
+            chip->w90 = 1;
+            break;
+        case 0x180:
+            chip->w92 = 1;
+            break;
+    }
 
     chip->w194 = chip->input.ext_as_in || chip->input.ext_lds_in || chip->input.ext_uds_in || (chip->input.ext_vaddress_in & 0x7fff80) != 0x50a000; // TMSS
     chip->w310 = !((chip->input.ext_vaddress_in & 0x7fff80) == 0x600000 && !chip->input.ext_as_in); // VDP range (goes to VSYNC pin)
 
-    chip->w119 = !(chip->input.ext_m3 && (chip->input.ext_vaddress_in & 0x7f8000) == 0x500000); // z80 ram
-
-    chip->w128 = !(chip->input.ext_m3 && (chip->input.ext_vaddress_in & 0x7fff80) == 0x508000); // IO chip
-
-    chip->w122 = !(chip->input.ext_m3 && (chip->input.ext_vaddress_in & 0x7ffe00) == 0x508800); // Z80 ctrl
-
-    chip->w129 = !(chip->input.ext_m3 && (chip->input.ext_vaddress_in & 0x7fff80) == 0x509000); // FDC
-
-    chip->w124 = !(chip->input.ext_m3 && (chip->input.ext_vaddress_in & 0x7fff80) == 0x509800); // TIME
-
-    chip->w136 = !((chip->input.ext_zaddress_in & 0xc000) == 0 && !chip->w220 && chip->input.ext_m3); // Z80 RAM
-
-    chip->w140 = !((chip->input.ext_zaddress_in & 0xe000) == 0x4000 && !chip->w220 && chip->input.ext_m3); // YM3438
-
-    chip->w150 = !(chip->input.ext_m3 && (chip->input.ext_zaddress_in & 0xff00) == 0x6000 && !chip->input.ext_zwr_in && !chip->w220); // z80 bank
-
-    chip->w86 = !((chip->input.ext_zaddress_in & 0xff00) == 0x7f00 && chip->input.ext_m3); // VDP
-
-    chip->ext_ia14 = !(chip->input.ext_m3 && chip->va14_in); // IA14
+    if (!chip->input.ext_m3)
+    {
+        chip->w119 = (chip->input.ext_vaddress_in & 0x7f8000) != 0x500000; // z80 ram
+        chip->w128 = (chip->input.ext_vaddress_in & 0x7fff80) != 0x508000; // IO chip
+        chip->w122 = (chip->input.ext_vaddress_in & 0x7ffe00) != 0x508800; // Z80 ctrl
+        chip->w129 = (chip->input.ext_vaddress_in & 0x7fff80) != 0x509000; // FDC
+        chip->w124 = (chip->input.ext_vaddress_in & 0x7fff80) != 0x509800; // TIME
+        chip->ext_ia14 = !chip->va14_in; // IA14
+        chip->w86 = (chip->input.ext_zaddress_in & 0xff00) != 0x7f00; // VDP
+        chip->w136 = !((chip->input.ext_zaddress_in & 0xc000) == 0 && !chip->w220); // Z80 RAM
+        chip->w140 = !((chip->input.ext_zaddress_in & 0xe000) == 0x4000 && !chip->w220); // YM3438
+        chip->w150 = !((chip->input.ext_zaddress_in & 0xff00) == 0x6000 && !chip->input.ext_zwr_in && !chip->w220); // z80 bank
+    }

```

Comment:
variables are not set when `ext_m3` is true

Reply:
Hm, i'm checking if its false, did i reverse the logic by accident?

Reply:
something like this I mean
```
if (ext_m3)
{
var = blah, blah;
}
else
{
var = 1;
}
```

Reply:
Ah i see, ill resolve the following issues.